docs/guide/resources/CLI.md

```diff
@@ -0,0 +1,99 @@
+```
+
+For example, if you have a file called `seed.csv` in the current directory, you would run the following command:
+
+```bash
+autolabel config seed.csv
+```
+
+## The `init` command
+If you would prefer to edit a json file directly, you can use the `init` command to generate a config file for you. To do this, simply run the following command:
+
+```bash
+autolabel init
+```
+
+By default, this will create a config file that looks like the one below: 
+
+```json
+{
+    "task_name": "[TODO] Enter task name",
+    "task_type": "[TODO] Enter task type",
+    "dataset": {
+        "delimiter": "[TODO] Enter delimiter",
+        "label_column": "[TODO] Enter label column name"
+    },
+    "model": {
+        "provider": "openai",
+        "name": "gpt-3.5-turbo",
+        "params": {}
+    },
+    "prompt": {
+        "task_guidelines": "[TODO] Enter task guidelines",
+        "example_template": "[TODO] Enter example template",
+        "few_shot_examples": "[TODO] Enter few shot examples",
+        "few_shot_selection": "[TODO] Enter few shot selection",
+        "few_shot_num": "[TODO] Enter few shot num"
+    }
+}
+```
+
+`init` will also take a seed file as an argument. Combined with other options, this can result in a very quick config file generation process. For example, if you have a file called `seed.csv` in the current directory, you could run the following command:
+
+```bash
+autolabel init seed.csv --task-name ToxicCommentClassification --task-type classification --delimiter , --label-column label --task-guidelines "You are an expert at identifying toxic comments." --example-template "Example: {example}\nLabel: {label}" --few-shot-examples seed.csv --few-shot-selection semantic_similarity --few-shot-num 5 --guess-labels
+```
+
+Resulting in the following config file for the civil comments dataset:
+
+```json
+{
+    "task_name": "ToxicCommentClassification",
+    "task_type": "classification",
+    "dataset": {
+        "delimiter": ",",
+        "label_column": "label"
+    },
+    "model": {
+        "provider": "openai",
+        "name": "gpt-3.5-turbo",
+        "params": {}
+    },
+    "prompt": {
+        "task_guidelines": "You are an expert at identifying toxic comments.",
+        "example_template": "Example: {example}\nLabel: {label}",
+        "few_shot_examples": "seed.csv",
+        "few_shot_selection": "semantic_similarity",
+        "few_shot_num": 5,
+        "labels": [
+            "not toxic",
+            "toxic"
+        ]
+    }
+}
+```
+
+## The `plan` command
+The `plan` command works identically to running `LabelingAgent({config}).plan({dataset})` in python. To use it, simply run the following command:
+
+```bash
+autolabel plan <path-to-dataset> <path-to-config>
+```
+
+## The `run` command
+The `run` command works identically to running `LabelingAgent({config}).run({dataset})` in python. To use it, simply run the following command:
+
+```bash
+autolabel run <path-to-dataset> <path-to-config>
+```
```

Comment:
it might not be possible to do this in the future (with the transform PR as you need to pass in the dataset to the agent function directly and the output is also a dataset) - not sure if we should keep this here. thoughts @Tyrest @nihit

Reply:
After the transform PR, we could still make this command work for autolabel by reading the csv in the CLI code instead and likewise writing a new csv for output. I've been using `plan` quite a bit to just quickly what an example prompt might look like from a config and `run` might be useful for those who want to do a quick run for ~100 rows.

Reply:
Good callout, yes the suggested change makes sense to me - let's do this in a followup PR now that transforms related PR is merged